Static/ComputeTheory.hs

```diff
@@ -102,19 +103,21 @@ computeLibEnvTheories le =
     in foldl upd Map.empty lns
 
 computeDGraphTheories :: LibEnv -> DGraph -> DGraph
-computeDGraphTheories le dgraph =
+computeDGraphTheories le dgraph = 
   let newDg = computeDGraphTheoriesAux le dgraph
   in groupHistory dgraph (DGRule "Compute theory") newDg
 
 
 computeDGraphTheoriesAux :: LibEnv -> DGraph -> DGraph
-computeDGraphTheoriesAux le dgraph =
+computeDGraphTheoriesAux le dgraph = let
+  tnodes = topsortedNodes dgraph 
+ in 
   foldl (\ dg l@(n, lbl) -> updatePending dg lbl
     (n, recomputeNodeLabel le dg l))
-     dgraph $ topsortedNodes dgraph
+     dgraph tnodes
```

Comment:
this is a mere reformulation. Is this needed?

Reply:
No (unless you prefer to have less $ and more names in use). I have added it for tracing. Reverted.